pkg/kubernetes/scripts/kube-client.js

```diff
@@ -442,13 +451,29 @@
                 var present = { };
                 var removed = { };
                 var i, len, frame, link, resource, key;
+                var cVersion, lVersion;
                 for (i = 0, len = drain.length; i < len; i++) {
                     resource = drain[i].object;
                     if (resource) {
                         link = resourcePath([resource]);
                         if (drain[i].type == "DELETED") {
                             delete objects[link];
                             removed[link] = resource;
+                        } else if (drain[i].checkResourceVersion) {
```

Comment:
That's the way I had it originally but then i changed it because I was a little worried relying on it so extensively should the assumptions above about resourceVersion not hold, since those guarantees aren't actually given in the kube docs. 

Also although  the etcd docs do say that their modifiedIndex is a unique, monotonically-incrementing integer. I know in go that uint64 does wraparound silently if you increment past the max and I wasn't sure how they handle that.

So I figured that by doing it like solves our current problem as long as the index is what we expect. And worse case where we get an unexpected resourceVersion value we are back to where we are now with races between handle and watches as opposed to potentially rejecting valid items from watches.

But maybe I'm just being too cautious. WDYT?

Do you think allowing handle calls to pass a param to skip the check is worth doing?